libs/sdk-ui-dashboard/src/model/state/layout/layoutReducers.ts

```diff
@@ -263,5 +278,22 @@ export const layoutReducers = {
     removeSectionItem: withUndo(removeSectionItem),
     replaceSectionItem: withUndo(replaceSectionItem),
     replaceWidgetHeader: withUndo(replaceWidgetHeader),
+    replaceWidgetDrills: withUndo(replaceWidgetDrill),
     undoLayout: undoReducer,
 };
+
+// reducer helpers
+
+const getWidgetByRef = (state: WritableDraft<LayoutState>, widgetRef: ObjRef) => {
```

Comment:
There is a selector for this `selectWidgetByRef` can we use that?

Reply:
I dont think so this function is helper method and is used in Immer scope. Originally SONAR force me to extract this into sep. method.

Reply:
Oh, ok, that makes sense.